patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2516,6 +2516,21 @@
  		}
  
  		public void DrawItems() {
+@@ -28382,7 +_,13 @@
+ 				Rain rain = Main.rain[i];
+ 				if (rain.active) {
+ 					value.X = rain.type * 4;
++
++					Texture2D tex = value2;
++					if (rain.waterStyle >= maxLiquidTypes) {
++						tex = LoaderManager.Get<WaterStylesLoader>().Get(waterStyle).GetRainTexture();
++					}
++
```

Comment:
Is there a way that declaring a new texture can be avoided, and instead `value2` reused? Just so that the patch below can be removed

Reply:
Just use `else { value2 = TextureAssets.Rain.Value }`, yeah, probably better.

Reply:
isn't `value2` already assigned to that? No need for the else then

Reply:
This happens for each loop iteration so it needs to be reverted.

Reply:
I.e. each particle of rain on the screen, which may be a mix of vanilla and modded.

Reply:
Oh, alright then